[NTOS:CC] Minor improvements

TAN-Gaming opened the pull request:
## Proposed changes
- Improve code consistency
- `CcRosLookupVacb`: Simplify loop exiting in case of the VACB was found
- Reduce 1 master lock acquire (in `CcRosGetVacb`)
- `CcRosTrimCache`: Remove a unnecessary inc-dec refcount pair in the loop. Replace it with a get refcount call.
- Use ReactOS naming style
- Remove useless DPRINTs
- Fix some comments

TAN-Gaming committed:
[NTOS:CC] Improve CcRosTrimCache

- Remove a unnecessary inc-dec refcount pair in the loop. Replace it with a get refcount call.
- Fix some comments

TAN-Gaming committed:
[NTOS:CC] Improve CcRosFreeOneUnusedVacb

- Improve code consistency
- Fix some comments

TAN-Gaming committed:
[NTOS:CC] Improve CcRosGetVacb, CcRosLookupVacb and CcRosCreateVacb

- CcRosGetVacb: Don't move the VACB to the tail of the LRU list in case of creating a new one. It's already at the tail of the list. This reduces 1 master lock acquire.
- CcRosLookupVacb: Simplify loop exiting in case of the VACB was found.
- Improve code consistency
- Remove useless DPRINTs
- Fix some comments

binarymaster reviewed (commented):
I think it would be better to fix naming style in separate commits, so other changes are visible better.

TAN-Gaming commented:
> I think it would be better to fix naming style in separate commits, so other changes are visible better.

OK.

TAN-Gaming committed:
[NTOS:CC][FORMATTING] Use ReactOS naming style

- CcRosTrimCache
- CcRosFreeOneUnusedVacb
- CcRosGetVacb
- CcRosLookupVacb
- CcRosCreateVacb

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -520,25 +519,27 @@ CcRosTrimCache(
         }
 
         /* Only keep iterating though the loop while the lock is held */
         current_entry = current_entry->Flink;
```
Let's suppose that the code with the comment `/* Page out the VACB */` above could be run (I know that it's currently in the disabled part of the `#if 1`, so it currently is NOT run...).
We add a reference to the `current` VACB then release all the locks in order to run that "Page out the VACB" code.
I could imagine that during that time, some other code could reduce the `current` VACB ref count down to 1 (is it possible?). Then we reacquire all locks and remove our reference acquired on `current`. In that hypothetical scenario, its reference is now == 0.
`current_entry` is coming from a field (`VacbLruListEntry`) within that `current` VACB, whose reference count is now zero.
Because `CcRosVacbDecRefCount` destroys the VACB when the refcount reaches zero (see [here](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/cc/view.c;hb=90de2b660b12919c2f60f2f31ad41e37c123d662#l75) or [there](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/include/internal/cc.h;hb=90de2b660b12919c2f60f2f31ad41e37c123d662#l497)), when we get here doing `current_entry = current_entry->Flink;`, it seems to me that we would now reference invalid memory.
The original code was going around this problem because we had the `current` VACB still referenced while getting the `current_entry->Flink`.

HBelusca replied:
Note that in `CcRosFreeOneUnusedVacb` (your second commit), the `current_entry->Flink` dereferencing is done _while it was certain that VACB was in the list_, so I think that in that case it is OK to use it.

TAN-Gaming replied:
The original code implies that freeing VACB on the `CcRosVacbDecRefCount` call shouldn't be possible here because freeing must be done outside of locks (see [a comment in CcRosFlushDirtyPages](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/cc/view.c;h=7c13afb3220f40a7d43f4ddbc79d2783a5c08483;hb=90de2b660b12919c2f60f2f31ad41e37c123d662#l405)).

TAN-Gaming replied:
After looking at it again, I decided to move it up before the "Page out the VACB" code for consistency. Thanks for pointing out.

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -520,25 +519,27 @@ CcRosTrimCache(
+            /* Unlink the VACB and mark for free */
             RemoveEntryList(&current->VacbLruListEntry);
             InitializeListHead(&current->VacbLruListEntry);
+            RemoveEntryList(&current->CacheMapVacbListEntry);
```
ok

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -757,38 +757,40 @@ CcRosFreeOneUnusedVacb(
             RemoveEntryList(&current->VacbLruListEntry);
             InitializeListHead(&current->VacbLruListEntry);
+            RemoveEntryList(&current->CacheMapVacbListEntry);
+            InitializeListHead(&current->CacheMapVacbListEntry);
```
ok

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -1005,22 +1024,10 @@ CcRosGetVacb (
-
-    OldIrql = KeAcquireQueuedSpinLock(LockQueueMasterLock);
-
-    /* Move to the tail of the LRU list */
```
Just so that I understand what the original purpose of the code was: objects at the _**tail**_ of the LRU list are actually those that have been the _**most**_-recently used?

TAN-Gaming replied:
From my understanding of the code, yes.

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -647,20 +648,31 @@ CcRosLookupVacb (
+    {
+        /* Move to the tail of the LRU list */
+        RemoveEntryList(&Ret->VacbLruListEntry);
+        InsertTailList(&VacbLruListHead, &Ret->VacbLruListEntry);
```
Question, to be sure: is this wishable to have `CcRosLookupVacb()` also move the found VACB to the tail of the LRU list, even when tht function is not called as part of `CcRosGetVacb()` ?

TAN-Gaming replied:
`CcRosLookupVacb` success means that the returned VACB is in use. It should be marked as most-recently used.

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -628,6 +628,7 @@ CcRosLookupVacb (
     PLIST_ENTRY current_entry;
     PROS_VACB current;
     KIRQL oldIrql;
+    PROS_VACB Ret = NULL;
```
```suggestion
    PROS_VACB current, Ret = NULL;
    KIRQL oldIrql;
```

TAN-Gaming replied:
I think the current style is more readable.

dt10812 replied:
> I think the current style is more readable.

@HBelusca yeah because usually NT design strongly discourages comma separated variable declarations for different structural roles and it's much cleaner

HBelusca commented on ntoskrnl/cc/view.c:
```diff
@@ -647,20 +648,31 @@ CcRosLookupVacb (
 
     KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);
+
+    if (Ret)
```
NOTE: Original code "equivalent" would have this `if (Ret) {...}` block just before the `KeReleaseSpinLockFromDpcLevel(&SharedCacheMap->CacheMapLock);`

TAN-Gaming replied:
The cache map lock is not needed for modifying the LRU list (see `CcRosCreateVacb`).

HBelusca commented:
(NOTE: Some of my comments are marked "outdated" because of the 4th commit "formatting style", but they would still apply, replacing the old variable names with new ones.)

tkreuzer commented on ntoskrnl/cc/view.c:
```diff
@@ -505,6 +504,9 @@ CcRosTrimCache(
             PFN_NUMBER Page;
 
+            /* Keep a reference on the VACB */
+            CcRosVacbIncRefCount(current);
```
Great idea, except that it only protects `current` and not `current_entry`, which points to the next entry, which is not referenced. (See comment above)

tkreuzer commented on ntoskrnl/cc/view.c:
```diff
@@ -741,14 +741,14 @@ CcRosFreeOneUnusedVacb(
     while ((current_entry != &VacbLruListHead) && (to_free == NULL))
     {
-        ULONG Refs;
         PROS_VACB current;
```
Move this as well?

tkreuzer commented on ntoskrnl/cc/view.c:
```diff
@@ -757,38 +757,40 @@ CcRosFreeOneUnusedVacb(
 
-            /* Reset it, this is the one we want to free */
-            RemoveEntryList(&current->CacheMapVacbListEntry);
-            InitializeListHead(&current->CacheMapVacbListEntry);
```
Any reason for this reorder?
